roles/openshift_metrics/tasks/main.yaml

```diff
@@ -1,85 +1,24 @@
-
-- name: Create temp directory for kubeconfig
-  command: mktemp -d /tmp/openshift-ansible-XXXXXX
+- name: Create temp directory for doing work in
+  command: mktemp -td openshift-metrics-ansible-XXXXXX
   register: mktemp
   changed_when: False
 
-- name: Record kubeconfig tmp dir
-  set_fact:
-    openshift_metrics_kubeconfig: "{{ mktemp.stdout }}/admin.kubeconfig"
+- debug: msg="Created temp dir {{mktemp.stdout}}"
+
+- name: Create temp directory for all our templates
+  file: path={{mktemp.stdout}}/templates state=directory mode=0755
+  changed_when: False
 
 - name: Copy the admin client config(s)
   command: >
-    cp {{ openshift_master_config_dir }}/admin.kubeconfig {{ openshift_metrics_kubeconfig }}
+     cp {{ openshift.common.config_base}}/master/admin.kubeconfig {{ mktemp.stdout }}/admin.kubeconfig
   changed_when: False
+  check_mode: no
+  tags: metrics_init
 
-- name: Set hosted metrics facts
-  openshift_facts:
-    role: hosted
-    openshift_env: "{{ hostvars
-                       | oo_merge_hostvars(vars, inventory_hostname)
-                       | oo_openshift_env }}"
-    openshift_env_structures:
-    - 'openshift.hosted.metrics.*'
-
-- set_fact:
-    # Prefer the master facts over bare variables if present, prefer
-    # metrics_public_url over creating a default using default_subdomain
-    metrics_hostname: "{{ openshift.hosted.metrics.public_url
-                          | default('hawkular-metrics.' ~ (openshift.master.default_subdomain
-                          | default(openshift_master_default_subdomain )))
-                          | oo_hostname_from_url }}"
-    metrics_persistence: "{{ openshift.hosted.metrics.storage_kind | default(none) is not none }}"
-    metrics_dynamic_vol: "{{ openshift.hosted.metrics.storage_kind | default(none) == 'dynamic' }}"
-    metrics_template_dir: "{{ openshift.common.config_base if openshift.common.is_containerized | bool else '/usr/share/openshift' }}/examples/infrastructure-templates/{{ 'origin' if deployment_type == 'origin' else 'enterprise' }}"
-    image_prefix: "{{ '-v IMAGE_PREFIX=' ~ openshift.hosted.metrics.deployer.prefix if 'prefix' in openshift.hosted.metrics.deployer else '' }}"
-    image_version: "{{ '-v IMAGE_VERSION=' ~ openshift.hosted.metrics.deployer.version if 'version' in openshift.hosted.metrics.deployer else '' }}"
-
-
-- name: Check for existing metrics pods
-  shell: >
-    {{ openshift.common.client_binary }}
-    --config={{ openshift_metrics_kubeconfig }}
-    --namespace openshift-infra
-    get pods -l {{ item }} | grep -q Running
-  register: metrics_pods_status
-  with_items:
-    - metrics-infra=hawkular-metrics
-    - metrics-infra=heapster
-    - metrics-infra=hawkular-cassandra
-  failed_when: false
-  changed_when: false
+- include: "{{role_path}}/tasks/install_metrics.yaml"
```

Comment:
`include install_metrics.yaml` should work here without having to use the `role_path` var.